drivers/powerp-txt.c

```diff
@@ -381,101 +385,97 @@ static void powpan_initinfo(void)
 static ssize_t powpan_status(status_t *status)
 {
 	ssize_t	ret;
+	ssize_t i;
+	ssize_t valid = 0;
+	int code = -1;
+	char value[32];
+	ssize_t ofs = 0;
+	char seen_i_freq = 0;
 
-	ser_flush_io(upsfd);
+	memset(status, 0, sizeof(status_t));
 
 	/*
 	 * WRITE D\r
 	 * READ #I119.0O119.0L000B100T027F060.0S..\r
 	 *      #I118.0O118.0L029B100F060.0R0218S..\r
-	 *      01234567890123456789012345678901234
-	 *      0         1         2         3
+	 *      #I118.1O118.1L13B100V27.5F60.0HF60.0R65Q1.4S\x80\x84\xc0\x88\x80W\r		(CST135XLU)
+	 *      01234567890123456789012345678901234567890123
+	 *      0         1         2         3         4
 	 */
-	ret = ser_send_pace(upsfd, UPSDELAY, "D\r");
-
-	if (ret < 0) {
-		upsdebug_with_errno(3, "send");
-		return -1;
-	}
-
-	if (ret == 0) {
-		upsdebug_with_errno(3, "send: timeout");
-		return -1;
-	}
-
-	upsdebug_hex(3, "send", "D\r", 2);
-
-	usleep(200000);
```

Comment:
Just in case: the `powpan_command()` codepath has a 100000 sleep in equivalent location. Should any problems be expected with the shortening from 200000 here? (Don't know - e.g. UPS controller may be slower than with other commands to collect the status info?)

Reply:
It's just a delay to allow the serial port to transfer the full response (2400 baud) before we read. It actually runs fine without any delay at all. *200000* is actually slightly long in my case, is closer to *175000* total time to read the response. So I guess it's a trade-off, waiting in `select()` and doing a little extra processing but finishing faster, or waiting in `usleep()`. I'm happy to add a parameter to that function to make it configurable if you prefer.